internal/provider/provider.go

```diff
@@ -123,7 +136,15 @@ func (p *TerraformCloudProvider) Configure(ctx context.Context, req provider.Con
 		allowInsecure = data.AllowInsecure.ValueBool()
 	}
 
-	client, err := client.NewConnectionWithAPIKey(endpoint, allowInsecure, apiKey)
+	clientVersion := "2023-10-01-00"
```

Comment:
TY for updating the clientVersion. Leaving the old version of the client is likely an oversight on our part. 

IMO, we should default this to the latest version and replace the older version.  It's not clear what breaks though if we update it.  Will you please update the default here to latest, run the test suite `make testacc` and let us know if anything breaks with that update?

Reply:
I updated the default version but original namespace tests with CA auth started failing with

```
    namespace_resource_test.go:53: Step 1/2 error: Error running apply: exit status 1
        
        Error: Failed to create namespace
        
          with temporalcloud_namespace.terraform,
          on terraform_plugin_test.tf line 6, in resource "temporalcloud_namespace" "terraform":
           6: resource "temporalcloud_namespace" "terraform" {
        
        must enable mTLS when a CA cert is provided on namespace creation
```

I am going to keep the default as old

Reply:
I think that can be resolved by adding `Enabled: true` in [this](https://github.com/temporalio/terraform-provider-temporalcloud/pull/130/files#diff-6782cd28bf6e98777610c007031a5c3fc554bcb19a08edb61a22258e34136e8eR304) block where the mtls spec is being set.

Reply:
sure, can do that (Though I feel like this PR is solving two things at once)

Reply:
Yes agreed, we could split out the version upgrade to its own PR.

Reply:
thanks - I will keep this PR in its current state then and open a new one with upgrade.

Though do you agree on having the option of supplying `client_version` to the provider?

Reply:
I am in favor of keeping the version hard coded on the provider as things stand today.
The version is tightly coupled with the protos. Since the protos are compiled into the provider, i prefer keeping the version compiled into the provider too.

Reply:
@anekkanti FYI - The protos are no longer compiled in the provider after #121

Reply:
I'll go ahead and update the version and remove `client_version` field

Reply:
updated as discussed

Reply:
@anekkanti thank you for upgrading the client. I have removed those changes from my PR and rebased